### DISCUSSION TITLE
Exclude code in __main__ from documentation

### DISCUSSION THREAD

**Discussion by hliebert** [mkdocs/mkdocs#3781] [Q&A]
Is there a way to exclude code under `if __name__ == "__main__":` from the documentation?

**Comment by pawamoy** [comment]
You're probably using mkdocstrings-python to document your code. Then yes, you could use these options:

- https://mkdocstrings.github.io/python/usage/configuration/members/#members
- https://mkdocstrings.github.io/python/usage/configuration/members/#filters

Or you could write a [Griffe extension](https://mkdocstrings.github.io/griffe/guide/users/extending/) that prunes the object tree :slightly_smiling_face:

**Reply by hliebert** [reply]
Thanks for the quick reply. I'm aware of those options. I'm using mkdocstrings-python and mkdocs-gen-files to generate the the documentation pages. I'm looking for a way to exclude everything defined in `__main__` without explicitly setting members/filters for each page.

**Reply by pawamoy** [reply]
You could refactor your modules as to not define anything public under the condition. However I understand this is not ideal: I think objects under this condition should be marked as "not existing at runtime" (in terms of API), like objects guarded under TYPE_CHECKING conditions, since such objects are indeed not available when importing the module (contrary to executing it with `python -m`). You can open a feature request if you want :)
